pkg/operator/podidentity/azurepodidentitywebhook.go

```diff
@@ -46,3 +49,8 @@ func (a AzurePodIdentity) GetImagePullSpec() string {
 func (a AzurePodIdentity) Name() string {
 	return "azure"
 }
+
+func (a AzurePodIdentity) ApplyDeploymentSubstitutionsInPlace(deployment *appsv1.Deployment, client client.Client, logger log.FieldLogger) error {
+	deployment.Spec.Template.Spec.Containers[0].Image = a.GetImagePullSpec()
```

Comment:
This is required for all platforms. Keep this in the controller code to reduce duplicate code. Keep these platform specific functions for platform specific changes.